#!/usr/bin/env python
PACKAGE = "robotx_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_area", 				int_t,    0, "An Integer parameter", 300,	0,10000)
gen.add("show_screen",			str_t,		0, "Choose which screen color to be shown", "red")
#blue
gen.add("blue_H_low", 			int_t,    0, "An Integer parameter", 105,	0,179)
gen.add("blue_H_high", 			int_t,    0, "An Integer parameter", 130,	0,179)
gen.add("blue_S_low",  			int_t,    0, "An Integer parameter", 50,	0,255)
gen.add("blue_S_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("blue_V_low",  			int_t,    0, "An Integer parameter", 50,	0,255)
gen.add("blue_V_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
#green
gen.add("green_H_low", 			int_t,    0, "An Integer parameter", 40,	0,179)
gen.add("green_H_high", 		int_t,    0, "An Integer parameter", 98,	0,179)
gen.add("green_S_low",  		int_t,    0, "An Integer parameter", 50,	0,255)
gen.add("green_S_high", 		int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("green_V_low", 			int_t,    0, "An Integer parameter", 50,	0,255)
gen.add("green_V_high",			int_t,    0, "An Integer parameter", 255,	0,255)
#yellow
gen.add("yellow_H_low", 		int_t,    0, "An Integer parameter", 20,	0,179)
gen.add("yellow_H_high", 		int_t,    0, "An Integer parameter", 30,	0,179)
gen.add("yellow_S_low",  		int_t,    0, "An Integer parameter", 80,	0,255)
gen.add("yellow_S_high", 		int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("yellow_V_low",  		int_t,    0, "An Integer parameter", 100,	0,255)
gen.add("yellow_V_high", 		int_t,    0, "An Integer parameter", 255,	0,255)
#red
gen.add("red_H_low1", 			int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("red_H_high1", 			int_t,    0, "An Integer parameter", 10,	0,179)
gen.add("red_H_low2", 			int_t,    0, "An Integer parameter", 165,	0,179)
gen.add("red_H_high2", 			int_t,    0, "An Integer parameter", 179,	0,179)
gen.add("red_S_low",  			int_t,    0, "An Integer parameter", 80,	0,255)
gen.add("red_S_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("red_V_low",  			int_t,    0, "An Integer parameter", 85,	0,255)
gen.add("red_V_high", 			int_t,    0, "An Integer parameter", 255,	0,255)
#white
gen.add("white_H_low", 			int_t,    0, "An Integer parameter", 0,		0,179)
gen.add("white_H_high", 		int_t,    0, "An Integer parameter", 255,	0,179)
gen.add("white_L_low",  		int_t,    0, "An Integer parameter", 100,	0,255)
gen.add("white_L_high", 		int_t,    0, "An Integer parameter", 255,	0,255)
gen.add("white_S_low", 			int_t,    0, "An Integer parameter", 0,		0,255)
gen.add("white_S_high",			int_t,    0, "An Integer parameter", 255,	0,255)
#black
gen.add("black_H_low", 			int_t,    0, "An Integer parameter", 117,	0,179)
gen.add("black_H_high", 		int_t,    0, "An Integer parameter", 140,	0,179)
gen.add("black_S_low",  		int_t,    0, "An Integer parameter", 40,	0,255)
gen.add("black_S_high", 		int_t,    0, "An Integer parameter", 245,	0,255)
gen.add("black_V_low", 			int_t,    0, "An Integer parameter", 4,		0,255)
gen.add("black_V_high",			int_t,    0, "An Integer parameter", 150,	0,255)

exit(gen.generate(PACKAGE, "robotx_vision", "detection"))
